crates/database/src/ethflow_orders.rs

```diff
@@ -47,26 +47,41 @@ pub async fn read_order(
 
 pub async fn refundable_orders(
     ex: &mut PgConnection,
-    id: i64,
+    min_valid_to: i64,
+    min_validity_duration: i64,
+    min_slippage: f64,
 ) -> Result<Vec<EthOrderPlacement>, sqlx::Error> {
     const QUERY: &str = r#"
-        SELECT * FROM ethflow_orders eo
-        LEFT JOIN trades t on eo.uid = t.order_uid
-        WHERE eo.valid_to < $1
-        AND eo.is_refunded = false
-        AND t.order_uid is null
+SELECT eo.uid, eo.valid_to, eo.is_refunded from orders o 
+INNER JOIN ethflow_orders eo on eo.uid = o.uid 
+INNER JOIN order_quotes oq on o.uid = oq.order_uid
+LEFT JOIN trades t on o.uid = t.order_uid
+WHERE 
+eo.is_refunded = false
+AND t.order_uid is null
+AND eo.valid_to < $1
+AND (1.0 - o.buy_amount * 1.0 / oq.buy_amount) > $3
```

Comment:
Technically `buy_amount` could be zero if the order was generated directly onchain. Probably we should also require that `buy_amount != 0`, otherwise I assume the entire query would fail.

Reply:
Also, I'm a bit surprised that the sell amount doesn't appear.

Reply:
For sell orders, the o.sell_amount == eo.sell_amount. This is ensured by the code that makes queries. And ethflow orders only have sell orders
I should drop a comment...